### PR TITLE
WIP Graphql poc

### PR DESCRIPTION
**Pull request by farmasek**
Created basic GraphQL implementation, regarding #1495  and  https://github.com/odota/ui/issues/1211

GraphQL api is available at `/api/graphql`
GraphiQL editor is running at `api/graphiql`

example of query is
image

Schema can be now extended of more features. 

The SQL offset might not be correct.

### CODE REVIEW

File: store/search.js [hunk 1]
```diff
@@ -14,6 +14,8 @@ function findPlayer(search, cb) {
 
 function search(options, cb) {
   const query = options.q;
+  const offset = options.page ? options.page * options.pageSize : 0;
+  const pageSize = options.pageSize || 500;
```
Comment (howardchung) [inline store/search.js]: The pageSize needs to be capped at a value (recently changed to 250) in order to prevent timeouts

**Comment by dartharaknus** [comment]
Looks great man. Ping me on discord if you want to discuss anything about graphql in odota.

File: routes/api.js [hunk 2]
```diff
@@ -79,5 +83,12 @@ Object.keys(spec.paths).forEach((path) => {
     api[verb](route(), func);
   });
 });
+api.use('/graphql', bodyParser.json(), graphqlExpress({ schema }));
```
Comment (howardchung) [inline routes/api.js]: can these be added to spec.js so they show up in the documentation?

Comment (farmasek) [reply routes/api.js]: there will be only one endpoint, specific queries are shown on GraphiQL, its something like interactive documentation

Comment (howardchung) [reply routes/api.js]: We should still list the graphql endpoint so visitors can see it exists in the docs

**Comment by sepehr500** [comment]
If this gets implemented, we can consider adding Apollo Client to the front end. I have had some great experiences with it. Would really clean up our code.

**Comment by farmasek** [comment]
My intention is to use Apollo on client side, but unfortunately i dont have time much time now, so i dont see it finished any time soon.

**Comment by sepehr500** [comment]
I would be interested in taking this over. I have implemented GraphQL on an enterprise system before.

**Comment by howardchung** [comment]
Sure, go for it!

**Comment by howardchung** [comment]
The idea is still sound, if someone wants to do add graphql support. That said, I don't know if we'd be rewriting the web UI to use graphQL, so it would be largely academic

**Pull request closed without merging**